packages/core/src/ChatEngine.ts

```diff
@@ -166,29 +166,78 @@ export class CondenseQuestionChatEngine implements ChatEngine {
   }
 }
 
+export interface Context {
+  message: ChatMessage;
+  nodes: NodeWithScore[];
+}
+
+export interface ContextGenerator {
```

Comment:
I'm not a fan of creating a separate class for generating context (in my opinion the response builder/retriever should suffice), but will merge this and push a new snapshot.